### DISCUSSION TITLE
Questions about Heat Transfer Calculation Boundaries

### DISCUSSION THREAD

**Discussion by Gallasd** [idaholab/moose#21387] [Q&A General] [answered]
I have a question, for a physics problem, suppose the surface of an object needs to impose two boundary conditions, such as radiation and heat flow boundary, although radiation is also a heat flow.
For radiation, I use FunctionRadiativeBC.For heat flow, and I use NeumannBC,but getting error of duplicate boundary.
Therefore, I consider whether it is necessary to build a function boundary to convert radiation into heat flow,then add them together to calculate. I try to use FunctionNeumannBC.But it is found that the function can only be related to space and time, not temperature.
So I don't know how to deal with this kind of problem that needs to impose two boundary conditions.Whether the kernel needs to be modified.
Thank you for reading, I hope you can help me.

**Comment by GiudGiud** [comment]
Hello

The Neumann BC is not meant to be combined, it must provide all of the flux. Use the convective heat flux BC instead
https://mooseframework.inl.gov/source/bcs/CoupledConvectiveHeatFluxBC.html
Guillaume

**Reply by Gallasd** [reply]
Hello
I'm sorry I didn't understand you,
CoupledConvectiveHeatFluxBC, as I understand it,  is possible to use vector coupling to compute the heat flux for multi-phase fluids.The heat flow here is actually q in the usual sense, not convection.I see the introduction of the boundary, the temperature is first order, which is not consistent with the calculation of radiation.
I hope you can provide me with a more detailed answer, thank you for your help

**Reply by GiudGiud** [reply]
Look at this input that has both on different boundaries. But seems to work to add radiation on the convective boundary
https://github.com/idaholab/moose/blob/next/modules/heat_conduction/test/tests/radiative_bcs/ad_function_radiative_bc.i

```
[BCs]
  [top_right]
    type = ADConvectiveHeatFluxBC
    variable = temp
    boundary = top_right
    T_infinity = 300.0
    heat_transfer_coefficient = 3.0
  []
  [bot_right]
    type = ADFunctionRadiativeBC
    variable = temp
    boundary = bot_right
    # htc/(stefan-boltzmann*4*T_inf^3)
    emissivity_function = '3/(5.670367e-8*4*300*300*300)'
  []
[]
```

**Comment by Gallasd** [comment, marked as answer]
hello
I'm glad you've been paying attention to this issue, because of some things I didn't reply to you right away, I'm very sorry

The study you provided should use a radiation study instead of convection by simplifying it. In the calculation example, the convection condition is applied to the top_right, and the simplified radiation is applied to the bot_right. I think this is still one boundary corresponding to one boundary condition, and it does not solve the problem of applying two working conditions to one boundary.

I checked all the heat-related boundaries provided by moose's official website and found that no similar boundaries are provided
![image](https://user-images.githubusercontent.com/107783970/175755249-eeec4d61-8aa0-445e-8a1d-218eefe74f18.png)

Among them, the fourth order corresponds to radiation, the first order corresponds to convection, and C corresponds to constant heat flux density.

**Reply by GiudGiud** [reply]
Well first it really woudlnt be hard to add this particular boundary condition for your needs. We're talking 10 minutes of coding max here. 

Secondly, while the input i linked does indeed use radiation and convection on different boundaries, you can add a second radiation condition on the same boundary as the convective one and it still works.

**Reply by Gallasd** [reply]
I solved the problem, one geometry bound can impose different bounds, thanks for your help
